modules/lldp/cli.c

```diff
@@ -0,0 +1,209 @@
+		req.tx = 1;
+	} else if (arg_str(p, "off") != NULL) {
+	} else {
+		printf("lldp enable API ERROR");
+		return CMD_ERROR;
+	}
+
+	if (gr_api_client_send_recv(c, GR_LLDP_ENABLE, sizeof(req), &req, &resp) < 0)
+		return CMD_ERROR;
+
+	return CMD_SUCCESS;
+}
+
+static cmd_status_t lldp_set_ttl(const struct gr_api_client *c, const struct ec_pnode *p) {
+	struct gr_lldp_set_ttl_req req;
+	void *resp;
+
+	if (arg_u16(p, "TTL", &req.ttl) < 0)
+		return CMD_ERROR;
+
+	if (gr_api_client_send_recv(c, GR_LLDP_SET_TTL, sizeof(req), &req, &resp) < 0)
+		return CMD_ERROR;
+
+	return CMD_SUCCESS;
+}
+
+static cmd_status_t lldp_show_config(const struct gr_api_client *c, const struct ec_pnode *p) {
+	struct gr_lldp_show_config_resp *resp = NULL;
+	void *resp_ptr;
+	(void)p;
+
+	if (gr_api_client_send_recv(c, GR_LLDP_SHOW_CONFIG, 0, NULL, &resp_ptr) < 0)
+		return CMD_ERROR;
+
+	resp = resp_ptr;
+	printf("lldp: rx: %s tx: %s ttl: %d\n",
+	       resp->rx ? "on" : "off",
+	       resp->tx ? "on" : "off",
+	       resp->ttl);
+
+	return CMD_SUCCESS;
+}
+
+static void decode_lldp(uint16_t data_len, uint8_t *data, char *ifname, uint8_t brief) {
+	uint16_t offset = 0;
+	char port_desc[512];
+	char sys_name[512];
+	char sys_desc[512];
+	uint16_t ttl = 0;
+
+	do {
+		uint8_t type = data[offset + 0] >> 1;
+		uint16_t len = ((data[offset + 0] & 0x1) << 8) + data[offset + 1];
+		offset += 2;
+		switch (type) {
+		case T_CHASSIS_ID:
+			break;
+		case T_PORT_ID:
+			break;
+		case T_TTL:
+			ttl = ntohs(*(uint16_t *)&data[offset]);
+			break;
+		case T_PORT_DESC:
+			memcpy(port_desc, &data[offset], len);
+			port_desc[len] = '\0';
+			break;
+		case T_SYSTEM_NAME:
+			memcpy(sys_name, &data[offset], len);
+			port_desc[len] = '\0';
+			break;
+		case T_SYSTEM_DESC:
+			memcpy(sys_desc, &data[offset], len);
+			port_desc[len] = '\0';
+			break;
+		case T_END:
+			goto end;
+			break;
+		default:
+			break;
+		}
+end:
+		offset += len;
+	} while (offset < data_len);
+
+	printf("%3ds   %s \t%s \t", ttl, ifname, sys_name);
+	if (brief)
+		printf("\n");
+	else
+		printf("%s \t%s \t%s\n", sys_name, sys_desc, port_desc);
+}
+
+static cmd_status_t lldp_show_neighbors(const struct gr_api_client *c, const struct ec_pnode *p) {
+	struct gr_lldp_show_neighbors_resp *resp = NULL;
+	struct gr_iface iface;
+	uint8_t brief = 1;
+	void *resp_ptr;
+	(void)p;
+
+	if (gr_api_client_send_recv(c, GR_LLDP_SHOW_NEIGH, 0, NULL, &resp_ptr) < 0)
+		return CMD_ERROR;
+
+	resp = resp_ptr;
+	printf("LLDP Neighbors: %d\n", resp->n_neigh);
+	printf("Age  TTL  ifname\tSys name\t");
+	if (brief)
+		printf("\n");
+	else
+		printf("sys desc\tport_desc\n");
+	for (uint16_t i = 0; i < resp->n_neigh; i++) {
+		clock_t age = (resp->now - resp->neighbors[i].last_seen) / 1000000;
+
+		if (iface_from_id(c, resp->neighbors[i].iface_id, &iface) < 0)
+			return CMD_ERROR;
+
+		printf("%3ld ", age);
+		decode_lldp(
+			resp->neighbors[i].n_tlv_data,
+			resp->neighbors[i].tlv_data,
+			iface.name,
+			brief
+		);
+	}
```

Comment:
Could you use libsmartcols to print the table here? See `show ip nexthop` for inspiration.

https://github.com/rjarry/grout/blob/bed15f4ef48ebf770d84fbb9f52be736c742ac06/modules/ip/cli/nexthop.c#L79-L126

Reply:
I'm rewriting this part indeed: for a verbose output, I intend to switch to an output similar to `lldpcli show neigh`:
```
-------------------------------------------------------------------------------
Interface:    ens1f0, via: LLDP, RID: 1, Time: 0 day, 00:00:09
  Chassis:
    ChassisID:    mac dc:2c:6e:47:35:77
    SysName:      crs317
    SysDescr:     MikroTik RouterOS 7.12.1 (stable) Nov/17/2023 11:38:45 CRS317-1G-16S+
    MgmtIP:       192.168.1.3
    MgmtIP:       2a01:cb19:141e:f200::3
    MgmtIP:       fe80::de2c:6eff:fe47:3587
    Capability:   Bridge, on
    Capability:   Router, on
  Port:
    PortID:       ifname bridge/sfp-sfpplus6
    TTL:          120
------------------------------------------------------------------------------- 

```

For the brief output, I'll indeed use libsmartcol.